init.ps1

```diff
@@ -81,6 +95,78 @@ function CreateDirectory($message, $path){
     New-Item -Force -ItemType Directory -Path $path | Out-Null
 }
 
+function Copy-GithubRepository
+{
+	param(
+	[Parameter(Mandatory=$True)]
+	[string] $Name,
+	
+	[Parameter(Mandatory=$True)]
+	[string] $Author,
+	
+	[Parameter(Mandatory=$False)]
+	[string] $Branch = "master",
+	
+	[Parameter(Mandatory=$False)]
+	[string] $GithubTokenVariableAssetName = "GithubToken",
+	
+	[Parameter(Mandatory=$True)]
+	[string] $sourceDirectory = "src"
+	
+	)
+	
+	$importRepoDirectory =  "$PWD\importedRepo"
+	CreateDirectory "Import Code Template Directory: $importRepoDirectory"  "$importRepoDirectory"
+	
+	$ZipFile = "$importRepoDirectory\$Name.zip"
+	$OutputFolder = "$importRepoDirectory\$Name"
+	$projectSourceDirectory = "$PWD\$sourceDirectory"
+   
+	$RepositoryZipUrl = "https://api.github.com/repos/$Author/$Name/zipball/$Branch"
+
+	# download the zip
+	Invoke-RestMethod -Uri $RepositoryZipUrl -OutFile $ZipFile
+	
+	# extract the zip
+	New-Item -Path $OutputFolder -ItemType Directory | Out-Null
+		
+	[System.Reflection.Assembly]::LoadWithPartialName('System.IO.Compression.FileSystem') | Out-Null
+	[System.IO.Compression.ZipFile]::ExtractToDirectory($ZipFile, $OutputFolder)

```

Comment:
I didn't try that.  I didn't want perfect to get in the way of good enough, but this could be a better longer term approach.